ossindex/internal/cache/cache.go

```diff
@@ -0,0 +1,168 @@
+//
+// Copyright 2020-present Sonatype Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+// Package cache has definitions and functions for processing the OSS Index Feed
+package cache
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/user"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/recoilme/pudge"
+	"github.com/sonatype-nexus-community/nancy/customerrors"
+	. "github.com/sonatype-nexus-community/nancy/logger"
+	"github.com/sonatype-nexus-community/nancy/types"
+)
+
+// DBName is exported from cache so that in tests we can switch the DB name, and avoid polluting our real DB
+var DBName = "nancy-cache"
+
+// TTL is exported so that you can a) set the TTL to a lower period of time if you want, and b)
+// so that in tests we can simulate an expired cache object
+var TTL = time.Now().Local().Add(time.Hour * 12)
```

Comment:
I would toss these into a struct and convert all the functions into methods of that struct.

```go
type Cache struct{
    DbName string
    TTL time.Duration
}

//instead of
func InsertValuesIntoCache(coordinates []types.Coordinate, ttl time.Time) (err error){

}

//it would be something like
func (c *Cache) Insert(coordinates []types.Coordinate, ttl time.Time) (err error){

}

```

Reply:
In: 45bf59a and 1d02418

Reply:
Something I don't like: you have to explicitly set the TTL and DBName, and I hate doing `New()` functions, but it's likely ok since this package is internal and for our use only.